model/stats/db.go

```diff
@@ -786,6 +796,34 @@ func (filter StatsFilter) buildMatchStageForTask() bson.M {
 		DbTestStatsIdProjectKeyFull:   filter.Project,
 		DbTestStatsIdRequesterKeyFull: bson.M{"$in": filter.Requesters},
 	}
+
+	// After cutting over the below build variants to use resmoke spawned
+	// with jasper (SERVER-54315), test names are now random UUIDs with
+	// with a human-readble display name. For the period between
+	// 09/01/21-09/13/21, the historical test stats calculations failed to
+	// for this (EVG-15396). We should ignore any test stats affected by
+	// this bug.
+	affectedVariants := []string{
+		"enterprise-rhel-80-64-bit",
+		"enterprise-rhel-80-64-bit-dynamic-required",
+	}
+	if len(filter.Tasks) == 0 || len(utility.StringSliceIntersection(filter.Tasks, affectedVariants)) > 0 {
```

Comment:
Why is this comparing the task filters against the buildvariant names? I would have thought it would compare against `filter.BuildVariants`.